### PR TITLE
Rejected issues are often not moved from Merge Ready to UAT Ready

### PR DESCRIPTION
**Pull request by justin-sadakhom**

### CODE REVIEW

**Review by caustin-ziro — CHANGES_REQUESTED** [review]

File: src/main/kotlin/github/GitHubClient.kt [hunk 1]
```diff
@@ -37,18 +41,42 @@ class GitHubClient : AutoCloseable {
...
         repoName: String = DEFAULT_GITHUB_REPOSITORY_NAME,
         branch: String,
+        numCommits: Int
     ): List<String> = runBlocking {
-        val query = GetBranchLogHistoryQuery(repoOwner, repoName, branch)
-        apolloClient.query(query).toFlow().single().data?.repository?.ref?.target?.onCommit?.history?.edges?.mapNotNull {
-            it?.node?.message
-        } ?: emptyList()
+        val commits = mutableListOf<String>()
+        var numPages = ceil(numCommits.toDouble() / MAX_COMMITS_IN_PAGE).toInt()
```
[the remark this answers is not in the document]
Comment (justin-sadakhom) [reply src/main/kotlin/github/GitHubClient.kt:51]: good call

File: src/main/kotlin/github/GitHubClient.kt [hunk 2]
```diff
@@ -37,18 +41,42 @@ class GitHubClient : AutoCloseable {
...
+        var hasNextPage = true
+        var cursor = Optional.absent<String>()
+        var commitsLeft = numCommits
+
+        while (numPages > 0 && hasNextPage) {
```
[the remark this answers is not in the document]
Comment (justin-sadakhom) [reply src/main/kotlin/github/GitHubClient.kt:56]: both conditions are needed

in the case where you query less than 1 page of commits (say 1 commit), the while loop would keep iterating until it hits the last page of commits even though it already has all the commits it needs

[the remark this answers is not in the document]
Comment (caustin-ziro) [reply src/main/kotlin/github/GitHubClient.kt]: I'm not sure I follow, but I trust you and this isn't a big deal. Approving!

**Review by caustin-ziro — APPROVED** [review]
lgtm, good work!

**Pull request merged**